tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/base/AbstractChouOrlandiOT.java

```diff
@@ -94,27 +95,24 @@ public StrictBitVector receive(boolean choiceBit) {
      * @return The random message received
      */
     private byte[] receiveRandomOt(boolean choiceBit) {
-        // S
-        T S = this.decodeElement(network.receive(otherId));
-        // T = G(S)
-        T T = hashToFieldElement(S, "FRESCO|ChouOrlandi|OT");
-        // R = T^c*g^x
-        // if c = 0 -> R = g^x
-        // if c = 1 -> R = T*g^x
+        // A
+        T A = this.decodeElement(network.receive(otherId));
+        // U = A^b*g^x
+        // if b = 0 -> U = g^x
+        // if b = 1 -> U = A*g^x
         BigInteger x = randNum.nextBigInteger(getDhModulus());
-        T R;
+        T U;
         if (choiceBit == false) {
-            R = T.exponentiation(BigInteger.ZERO).groupOp(getGenerator().exponentiation(x));
+            U = A.exponentiation(BigInteger.ZERO).groupOp(getGenerator().exponentiation(x));
         } else {
-            R = T.exponentiation(BigInteger.ONE).groupOp(getGenerator().exponentiation(x));
+            U = A.exponentiation(BigInteger.ONE).groupOp(getGenerator().exponentiation(x));
         }
-        network.send(otherId, R.toByteArray());
+        network.send(otherId, U.toByteArray());
```

Comment:
We still need to ensure that `toByteArray` always returns an array of the same amount of bytes regardless of the size of the underlying element. If not and `toByteArray` returns things as big endian we end up in situation where two different values will have the same byte representation.

Reply:
The underlying BigInteger implementation always returns a big endian representation. The EC implementation an encoded non-compressed point. The first one has no guarantees in the term of byte length, but the second will depend directly on the field size. However I am not quite sure how a variable byte length would be a problem? In any case should we then pad the sent bytes?

Reply:
Yes, padding should be added to ensuring equal byte length for all numbers

Reply:
Actually, I'm not sure this is a problem since the byte-representation is prepended with the sign in the first byte.

Reply:
You are right that for network communication it does not matter how many bytes is used.
However when we need to use the result we can get into issues if we  have a number that is represented using a 31 byte big endian integer, and we need to hash this. Thus a bunch of 0-bytes gets appended. Now the result will represent a much larger number than it actually is. In that case the number should be prepended with 0-bytes to ensure consistent encoding once it gets mapped to a 32 byte array (or something larger)

Reply:
Solved by 0375fd3